src/App.js

```diff
@@ -25,44 +24,97 @@ const generateSquares = () => {
   return squares;
 }
 
-const App = () => {
+const updateCurrentPlayer = (currentPlayer, setCurrentPlayer) => {
+  const updatePlayer = currentPlayer === PLAYER_1 ? PLAYER_2:PLAYER_1
+  setCurrentPlayer(updatePlayer)
+}
 
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
+const App = () => {
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1)
```

Comment:
I see that you are in the drop-semicolon camp. There are teams that prefer one style or the other, so be sure to adopt the practices of your team. I am firmly in the pro-semicolon camp myself. There are situations in JS (uncommon, but they exist) where leaving off the semicolons leads to improper parsing. It _can_ be a pain to include semicolons, especially when JS attempts to insert them on our behalf, but since it generally follows the same semicolon rules as languages like C and Java, it can be helpful to get used to using them here.

Alternatively, if you take a detour and do some work with C or Java, it can make it much clearer where semicolons belong in JS, since those other languages will absolutely complain about every missed semicolon! 😆